web/src/main/webapp/xslt/common/index-utils.xsl

```diff
@@ -60,7 +60,7 @@
     <xsl:param name="value" as="xs:string?"/>
     <xsl:value-of select="if ($value castable as xs:date
                           or $value castable as xs:dateTime
-                          or matches($value, '[0-9]{4}(-[0-9]{2})?'))
+                          or matches($value, '^[0-9]{4}$|^[0-9]{4}-[0-9]{2}$'))
```

Comment:
This still allow non valid dates, like `2021-40`. This only allow valid combinations for the month:
```
^[0-9]{4}$|^[0-9]{4}-(0[1-9]|1[012])$
```
This one allows months between 01 and 12 and reject the rest.